tools/clang/lib/SPIRV/CapabilityVisitor.cpp

```diff
@@ -718,6 +718,10 @@ bool CapabilityVisitor::visit(SpirvExecutionMode *execMode) {
     addExtension(Extension::EXT_shader_stencil_export,
                  "[[vk::stencil_ref_less_equal_back]]", execModeSourceLocation);
     break;
+  case spv::ExecutionMode::MaximallyReconvergesKHR:
+    addExtension(Extension::KHR_maximal_reconvergence, "",
```

Comment:
Is the recommended approach to unconditionally add extensions and then use the capability trimming pass? I assume that would work here.

Reply:
Yup, we should be able to remove it if the execution mode is not used.

Reply:
Yeah, I wanted to do that, as it seems natural, but the issue is: MaximallyReconvergesKHR is not a capability, nor requires one.
It's only an operand and an extension.

The way the capability trimming pass is built is it relies on capabilities first.
- we have a list of allowed capabilities to trim.
- we figure out what capabilities from that list can be removed.
- we list the extensions linked to those capabilities.
- we trim the capabilities & extensions found in the previous steps.

For this to be trimmed by the pass, the pass would need to be changed:
 - we'd need to also have a list of extensions we are allowed to remove
 - and a way to link operands usage to extension requirement.